2230/3/rap-battle/rap-battle.rb

```diff
@@ -0,0 +1,160 @@
+# p opts
+
+class FileReader
+  attr_reader :content
+
+  def initialize
+    @content = ''
+  end
+
+  def read(files)
+    @content = get_all_text(files)
+  end
+
+  # :reek:FeatureEnvy
+  # :reek:NestedIterators
+  # :reek:TooManyStatements
+  def get_all_text(files)
+    content = ''
+    files.each do |file_name|
+      fh = open "data/#{file_name.gsub(/:[()"'*.-]/) { |sym| '\\' + sym }.chomp}"
+      content += fh.read
+
+      fh.close
+    end
+    content
+  end
+end
+
+class DataLoader
+  attr_reader :data
+
+  def initialize
+    @data = [] # public
+    @rapper_names = [] # private
+  end
+
+  def load_data
+    @rapper_names = load_rapper_names
+    @data = build_data
+  end
+
+  private
+
+  def load_rapper_names
+    fh = open 'config/rapper_names'
+    fh.map(&:rstrip)
+  end
+
+  # Generate list of files
+  # rubocop:disable Style/InverseMethods
+  def build_file_list(folder)
+    Dir.entries(folder).select { |file_name| !File.directory? file_name }
+  end
+  # rubocop:enable Style/InverseMethods
+
+  # parse list of files
+  # :reek:TooManyStatements
+  def get_file_list(name)
+    common_file_list = build_file_list('data')
+    file_list = []
+    # filter rapper names by name
+    names = Utils.escape_string(@rapper_names.grep(/#{Utils.escape_string(name)}/)[0].gsub(', ', '|'))
+    common_file_list.grep(/#{names}/i).each do |file_name|
+      file_list << "#{file_name}\n" if file_name.sub(/\s(прот|vs|VS ).+/, '').index(/#{names}/)
+    end
+    file_list
+  end
+
+  # rubocop:disable Metrics/MethodLength
+  # rubocop:disable Metrics/AbcSize
+  # :reek:TooManyStatements
+  def build_data
+    @rapper_names.each do |rapper_name|
+      rapper_name = rapper_name.sub(/,.*/, '')
+
+      files = get_file_list(rapper_name)
+      text = FileReader.new.read(files)
+      battles_count = files.size
+      words_count = text.scan(/\S+/).size
+      bad_words_count = RussianObscenity.find(text).size
+      words_per_battle = (bad_words_count.fdiv battles_count).round(2)
+      words_per_stage = ((words_count.fdiv battles_count).fdiv 3).round(0)
+      # rubocop:disable Metrics/LineLength
+      data << { name: rapper_name, text: text, battles_count: battles_count, words_count: words_count, bad_words_count: bad_words_count, words_per_battle: words_per_battle, words_per_stage: words_per_stage }
+      # rubocop:enable Metrics/LineLength
+    end
+    data
+  end
+  # rubocop:enable Metrics/MethodLength
+  # rubocop:enable Metrics/AbcSize
+end
+
+class Report
+  def print
+    raise NotImplementedError
+  end
+end
+
+# :reek:Attribute
+class TopBadWordsReport < Report
+  attr_writer :top_bad_words
+
+  def initialize(top_bad_words)
+    @top_bad_words = top_bad_words
+  end
+
+  def print
+    print_top_bad_words(@top_bad_words)
+  end
+
+  private
+
+  # rubocop:disable Metrics/MethodLength
+  # rubocop:disable Metrics/AbcSize
+  # :reek:FeatureEnvy
+  # :reek:TooManyStatements
+  def print_top_bad_words(top_bad_words)
```

Comment:
too big
you shiuld take a look https://github.com/tj/terminal-table

Reply:
Thx, i have dreamed about it